packages/contracts/contracts/optimistic-ethereum/OVM/accounts/OVM_ECDSAContractAccount.sol

```diff
@@ -66,66 +59,46 @@ contract OVM_ECDSAContractAccount is iOVM_ECDSAContractAccount {
             bytes memory
         )
     {
-        bool isEthSign = _signatureType == Lib_OVMCodec.EOASignatureType.ETH_SIGNED_MESSAGE;
+        // Attempt to decode the transaction.
+        Lib_EIP155Tx.EIP155Tx memory transaction = Lib_EIP155Tx.decode(
+            _encodedTransaction,
+            Lib_ExecutionManagerWrapper.ovmCHAINID()
+        );
 
         // Address of this contract within the ovm (ovmADDRESS) should be the same as the
         // recovered address of the user who signed this message. This is how we manage to shim
         // account abstraction even though the user isn't a contract.
-        // Need to make sure that the transaction nonce is right and bump it if so.
         require(
-            Lib_ECDSAUtils.recover(
-                _transaction,
-                isEthSign,
-                _v,
-                _r,
-                _s
-            ) == address(this),
+            transaction.sender() == address(this),
```

Comment:
Notably the way this code path is currently done is that we re-encode the raw RLP within `sender()` to do recovery.  We could instead recover the sig based on `hash(msg.data[4:end])`.

A quick aside: can you briefly talk about the reasoning for making the `OVM_ECDSAContractAccount` accept encoded `bytes` now as opposed to just a single `Lib_EIP155Tx.EIP155Tx` argument?  Doesn't seem like it makes l2geth any different, and could add redundant encoding/decoding.  Just curious to hear it laid out; makes total sense in the `OVM_SequencerEntryPoint`.

Reply:
Hmmm yeah interesting, I hadn't thought of passing the `EIP155Tx` directly. I think that's a great idea. Will do this!

Reply:
Ok looking at this now, this introduces enough of a diff that we may want to leave it for another PR + treat it as an optimization if this method is too expensive.

Reply:
Yeah, it's a bit of a tradeoff.  Ultimately the most gas-efficient version could just require `extcodehash(msg.sender) == SEQUENCER_ENTRYPOINT_CODEHASH` and trust that code path explicitly instead of redoing the work.  Down to leave for future work.